app/server/src/main/scala/org/bitcoins/server/Main.scala

```diff
@@ -165,12 +165,19 @@ object Main extends App {
       wallet.processBlock(block)
       ()
     }
+    lazy val onHeaders: OnBlockHeadersReceived = { headers =>
+      wallet.updateTxoStatesFromHeader(headers.last)
```

Comment:
Also, `headers` can be empty. `bitcoind` returns an empty `headers` message when a fully synced node send `getheaders` command.